atomic_reactor/plugins/pre_koji_parent.py

```diff
@@ -226,12 +231,32 @@ def manifest_list_entries_match(self, image, build_id):
             v2_digest = archive['extra']['docker']['digests'][v2_type]
             koji_archives_data[arch] = v2_digest
 
-        if koji_archives_data == manifest_list_data:
-            self.log.info('Deeper manifest list check verified v2 manifest references match')
-            return True
-        self.log.warning('Manifest list refs "%s" do not match koji archive refs "%s"',
-                         manifest_list_data, koji_archives_data)
-        return False
+        platform_to_arch_dict = get_platform_to_goarch_mapping(self.workflow)
+        architectures = []
+        for platform in self.platforms:
+            architectures.append(platform_to_arch_dict[platform])
+
+        if not all(arch in koji_archives_data for arch in architectures):
+            self.log.warning('Requested platforms "%s" are missing in Koji archives "%s"',
+                             architectures, koji_archives_data)
+            return False
+
+        # manifest lists can be manually pushed to the registry to make sure a specific tag
+        # (e.g., latest) is available for all platforms.
+        # In such cases these manifest lists may include images from different koji builds.
+        # We only want to check the digests for the images built in the current parent koji build
+        err_msg = 'Manifest list digest %s differs from Koji archive digest %s for platform %s'
+        unmatched_digests = False
+        for arch in architectures:
+            if manifest_list_data[arch] != koji_archives_data[arch]:
```

Comment:
As the comment mentions, manifest lists can be manually pushed to the registry, is there a mechanism outside OSBS to ensure the arches included in the pushed manifest lists must match or contain arches defined in reactor config `platform_descriptor`? If no, `manifest_list_data[arch]` could raise `KeyError`. Is this an expected special case?

Reply:
@rcerven ^

Reply:
if there would be wrong arches in manifest_list, it would fail already in pull_base_image plugin

Reply:
Thanks robert